### PR TITLE
Input Verification Fix

### PR DESCRIPTION
**Pull request by bautrey37**
MSG-75

### CODE REVIEW

**Comment by bautrey37** [comment]
A general question: Should the `Gemfile.lock` be committed to the repo? As seen in the dry-validation version bump, it is hard to see how this affects all the dependencies. All the other repos commit this file.

**Comment by bautrey37** [comment]
Used by this commit: https://github.com/salemove/engagement-registry/pull/1536/commits/ea98299e787fd516601f6597485083086568d0a7

**Review by riiwo — COMMENTED** [review]
Please also bump the gem version (At least minor I guess, as this is a breaking change?)

**Comment by SteamRiC** [comment]
> A general question: Should the `Gemfile.lock` be committed to the repo? As seen in the dry-validation version bump, it is hard to see how this affects all the dependencies. All the other repos commit this file. CC @SteamRiC

This seems to be a hot topic in the ruby community. As salestation is shared, maybe you can ask #techmovers for opinions.
Although given that our other gems also have Gemfile.lock, I think it's best to be consistent here.

File: lib/salestation/app/input_verification.rb [hunk 1]
```diff
@@ -7,10 +7,10 @@ def verify_input(schema)
         -> (request) do
           result = schema.call(request.input)
           if result.success?
-            request.replace_input(result.output)
+            request.replace_input(result.to_h)
           else
             Deterministic::Result::Failure(
-              Errors::InvalidInput.new(errors: result.errors, hints: result.hints)
+              Errors::InvalidInput.new(errors: result.errors.to_h, hints: result.hints.to_h)
```
Comment (SteamRiC) [inline lib/salestation/app/input_verification.rb:13]: I think we may want to make this compatible with the older version as we have done with dry-types https://github.com/salemove/salestation/pull/49/commits/7b835c6f743b68a34be8d5c670417a6615972fe5

the upgrade is not easy and other ruby services still use the old version of dry validation

Comment (bautrey37) [reply lib/salestation/app/input_verification.rb:13]: glia-errors shows an example of the dry-validation version switching. https://github.com/salemove/glia-errors/blob/master/ruby/lib/glia/errors.rb#L12-L19

Comment (bautrey37) [reply lib/salestation/app/input_verification.rb:13]: Let me check which services are still on the old version.

Comment (bautrey37) [reply lib/salestation/app/input_verification.rb]: Added switching https://github.com/salemove/salestation/compare/86d30a08aa975c8002eebeac4081716828280fc9..282ff143a3cdc4367f4d41697dea384b8a184312

File: lib/salestation/app/input_verification.rb [hunk 2]
```diff
@@ -6,12 +6,27 @@ module InputVerification
       def verify_input(schema)
         -> (request) do
           result = schema.call(request.input)
-          if result.success?
-            request.replace_input(result.output)
+
+          dry_validation_version = Gem.loaded_specs['dry-validation'].version
+          if dry_validation_version < Gem::Version.new('1.0')
```
Comment (SteamRiC) [inline lib/salestation/app/input_verification.rb:11]: shouldn't this be `< Gem::Version.new('1.7')` as previously we expected dry validation to be `1.6`?

Comment (bautrey37) [reply lib/salestation/app/input_verification.rb:11]: The breaking change is across v1.0, not v1.6. idk why it wasn't changed before on the other upgrades, maybe the other services didn't use salestation like how engagement-api does. I only bumped dry-validation to fix a warning in the project.

**Review by SteamRiC — APPROVED** [review]

**Comment by bautrey37** [comment]
https://salemove.slack.com/archives/CBJJE330U/p1650961264956719

There is not a consensus on what is the right way yet regarding the Gemfile.lock

**Comment by SteamRiC** [comment]
I think it's fine not to add it in this PR as it only affects development of the gem and doesn't block this PR.

**Comment by bautrey37** [comment]
I think this repo publishes automatically when merged, so i'll just merge.

**Comment by bautrey37** [comment]
https://rubygems.org/gems/salestation/versions/4.6.0

**Pull request merged**